\documentclass[aspectratio=169,9pt,handout]{beamer}

%\transdissolve[duration=0.2] % Only works with Adobe Acrobat

% Some important packages
\usepackage{epstopdf}
\usepackage{ulem} % sout
\hypersetup{colorlinks=false, allcolors=purple}
\usepackage{booktabs}
\linespread{1.3}
\usepackage{tabularx,multirow}
\usepackage{makecell} % For makecell within tables
\usepackage{geometry}
\usepackage{fancybox}
\usepackage{algorithm2e}
\usepackage{amsmath, amssymb}
\input{notation-thesis.tex}

% New notation ------------------------------------------------------------------
\newcommand{\expo}{E}
\newcommand{\Expo}{\mathbf{\expo{}}}
\newcommand{\expod}[1]{\expo_{#1}^D}
\newcommand{\expoc}[1]{\expo_{#1}^C} 
\newcommand{\expoi}[1]{\expo_{#1}^I}

\newcommand{\covar}{\mathbf{x}}
\newcommand{\Covar}{\mathbf{X}}
\newcommand{\latent}{\mathbf{z}}
\newcommand{\Latent}{\mathbf{z}}

\newcommand{\Event}{\mbox{P}}
\newcommand{\pset}[1]{\mathcal{P}\left(#1\right)}
\newcommand{\size}[1]{\left|#1\right|}

\newcommand{\logit}[1]{\mbox{logit}^{-1}\left(#1\right)}
% -------------------------------------------------------------------------------

\usepackage[style=authoryear-comp]{biblatex}
\addbibresource{bibliography-aphylo.bib}
\addbibresource{bibliography-ergmito.bib}
\addbibresource{bibliography-extra.bib}
\addbibresource{bibliography.bib}
% \renewcommand{\bibsection}{\subsubsection*{\bibname } }

% Neat trick to change the mode mid-presentation
% https://tex.stackexchange.com/questions/91691/beamer-set-mode-mid-presentation?rq=1
\makeatletter
\newcommand\changemode[1]{%
	\gdef\beamer@currentmode{#1}}
\makeatother


% Styles
\usepackage{xcolor}
\usepackage{colortbl}
\definecolor{suffstat}{RGB}{10,159,0}
\definecolor{normconst}{RGB}{87,38,231}
\setbeamercolor{conclusions}{bg=usclightgray!60!white, fg=uscdarkgray}

% Noice!
\usetheme{usckeck}

\title[Stat. Comp. for Complex Systems]{%
	Triads, Dyads, and Gene Functions: When Social Network Analysis meets Phylogenetics
}
\author[GGVY -- \href{vegayon@usc.edu}{vegayon@usc.edu}]{George G Vega Yon, Ph.D.\\(with Paul D Thomas \and Paul Marjoram \and Huaiyu Mi \and Duncan Thomas \and John Morrison)}
\institute[USC-Keck]{\normalsize University of Southern California\\Department of Population and Public Health Sciences}
\date{Networks 2021\\{\footnotesize(virtual)}\\July 8, 2021\vspace{-1cm}}

% Some definitions
\def\cursection{\frame{\frametitle{Contents}\tableofcontents[current]}}
\newcommand{\ergmpkg}[0]{\texttt{ergm}}
\newcommand{\ergmitopkg}[0]{\texttt{ergmito}}
\newcommand{\aphylopkg}[0]{\texttt{aphylo}}
\graphicspath{{.}{fig/}{fig-whoami/}{fig-struct-test/}}

\usepackage{media9}


% ------------------------------------------------------------------------------
% ------------------------------------------------------------------------------
% --------------------------- END OF PREAMBLE ----------------------------------
% ------------------------------------------------------------------------------
% ------------------------------------------------------------------------------
\setbeamertemplate{note page}[plain]
%\setbeameroption{show notes}
\usepackage{pgfpages}
% \setbeameroption{show notes on second screen}

\newcommand{\hlc}[2]{{\only<#1>{\cellcolor{gray!50}#2}}}
\newcommand{\nhlc}[2]{{\only<#1>{#2}}}
\newcommand{\hlcAlt}[2]{\alt<#1>{\cellcolor{gray!50}#2}{#2}}

\newcommand{\byside}[3]{\begin{minipage}[t]{#1\linewidth}%
		\bigskip
		\centering%
		\shadowbox{\Large #2}\hfill\\\bigskip%
		#3%
\end{minipage}}

\usepackage{listings}

\definecolor{codegreen}{rgb}{0,0.6,0}
\definecolor{codegray}{rgb}{0.5,0.5,0.5}
\definecolor{codepurple}{rgb}{0.58,0,0.82}
\definecolor{backcolour}{rgb}{0.95,0.95,0.92}

\lstdefinestyle{mystyle}{
	backgroundcolor=\color{backcolour},   
	commentstyle=\color{codegreen},
	keywordstyle=\color{magenta},
	numberstyle=\tiny\color{codegray},
	stringstyle=\color{codepurple},
	basicstyle=\ttfamily\footnotesize,
	breakatwhitespace=false,         
	breaklines=true,                 
	captionpos=b,                    
	keepspaces=true,                 
	numbers=left,                    
	numbersep=5pt,                  
	showspaces=false,                
	showstringspaces=false,
	showtabs=false,                  
	tabsize=2
}

\lstset{style=mystyle}

\begin{document}

% ------------------------------------------------------------------------------
\begin{frame}%[noframenumbering]
\maketitle
\end{frame}

%% ------------------------------------------------------------------------------
%\begin{frame}
%\frametitle{What motivates my research}
%
%\centering
%
%\begin{center}
%\large
%\textcolor{usccardinal}{
%Essays on Bioinformatics and Social Network Analysis
%\linebreak{\small Statistical and Computational Methods for Complex Systems}
%}
%\end{center}\pause
%
%\begin{itemize}
%\item We live in a non-{\it IID} world.\pause
%\item Sometimes, we cannot understand a process unless we look at it as a whole.\pause
%\item But, while ``the whole'' may be difficult to analyze...\\
%\hfill \pause{\it modern} statistical and computational tools cap help us cope with that.
%\end{itemize}
%\end{frame}

%\begin{frame}[c]
%	\frametitle{Funding and support}
%	\begin{tabular}{m{.35\linewidth}<\centering m{.55\linewidth}}
%		\includegraphics[height=1.2cm]{nci-logo.pdf} & National Cancer Institute Grant \#1P01CA196596.\\ \\
%		\includegraphics[height=2cm]{aro-logo.png} & U.S. Army Research Laboratory and the U.S. Army Research Office under grant number W911NF-15-1-0577.
%	\end{tabular}
%	
%	\vfill
%	
%	\begin{center}
%		\includegraphics[height=1cm]{carc-logo.png}%\\USC's Center for Advance Research Computing (carc.usc.edu).
%	\end{center}
%	
%\end{frame}

\frame{\frametitle{Contents}
	\tableofcontents
	\vfill
	You can download the slides from\large{} {\color{usccardinal}\href{https://ggv.cl/slides/networks2021}{ggv.cl/slides/networks2021}}\normalsize
}


\section{The problem of genes' functions}

\begin{frame}[t]
	\usebeamertemplate{section intro}{}{}
	\textcolor{uscgold}{
		\Large {\bf The problem of genes' functions}
	}
\end{frame}

% ------------------------------------------------------------------------------

\begin{frame}[c]
%	\frametitle{Uncovering the role of genes}
	
	\Large Is gene \textit{XYZ} involved in process \textit{ABC}?\normalsize\bigskip
	
	\begin{minipage}[t]{.33\linewidth}
		\centering
		\includegraphics[width=1\linewidth]{aphylo-data-0.png} \\
		Complex to directly assess
	\end{minipage}\hfill
	\uncover<2->{\begin{minipage}[t]{.33\linewidth}
		\centering
		\includegraphics[width=1\linewidth]{aphylo-data-1.png}\\
		But we may know from other species
	\end{minipage}}\hfill
	\uncover<3->{\begin{minipage}[t]{.33\linewidth}
		\centering
		\includegraphics[width=1\linewidth]{aphylo-data-2.png}\\
		And we further know how these \textit{genetically} connected
	\end{minipage}}\hfill
	
\bigskip\uncover<4>{\raggedleft\Large ... let's rephrase the question. \normalsize}

\end{frame}

\begin{frame}[c,label=aphylo-prob-diagram]
	\begin{center}
		\normalsize Is the human gene \textbf{XYZ} involved in process \textbf{ABC}, \uline{given what we know about that for other \textit{related} species}?
	\end{center}
	
	\begin{figure}
		\includegraphics[width=.9\linewidth]{aphylo-data-probability.pdf}
	\end{figure}

\end{frame}

\begin{frame}[c,label=geneontology]
	\frametitle{The Gene Ontology Project}
	\begin{minipage}[m]{.33\linewidth}
		\includegraphics[width=1\linewidth]{aphylo-data-2.png}
	\end{minipage}\hfill
	\begin{minipage}[m]{.66\linewidth}
		\begin{figure}
			\includegraphics[width=.5\linewidth]{go-logo.png}
		\end{figure}
		\begin{itemize}
			\item<2-> $\sim$ 15,000 phylogenetic trees		
			\item<3-> $\sim$ 8 million annotations
			\item<4-> $\sim$ 600 thousand on human genes
			\item<5-> $\sim$ $<$ 10\% are based on experimental evidence... \uncover<6->{Improving our knowledge on genetics is fundamental for advancing Biomedical Research}
		\end{itemize}
	\end{minipage}
	\vfill \hfill
	\Large \uncover<7->{Only on 2020, 2,000+ COVID-19 papers using the GO \href{https://scholar.google.com/scholar?as\_ylo=2020\&q="gene+ontology"+(covid+OR+coronavirus)}{(Google Scholar)}}\normalsize\\
	\hfill\hyperlink{go-functions-types}{\beamergotobutton{more}}
	
\end{frame}

\begin{frame}[label=phylo-legend]
	\begin{figure}
		\mode<beamer>{%
			\only<1>{\includegraphics[width=.8\linewidth]{phylo-tree-legend-1.pdf}}%
			\only<2>{\includegraphics[width=.8\linewidth]{phylo-tree-legend-2.pdf}}%
			\only<3->{\includegraphics[width=.8\linewidth]{phylo-tree-legend.pdf}}%
		}%
		\mode<handout>{\includegraphics[width=.8\linewidth]{phylo-tree-legend.pdf}}
	\end{figure}

\vfill\hfill\hyperlink{aphylo-good}{\beamergotobutton{example}}

\end{frame}




%--------------------------------------------------------------------------------
\newcommand{\oinclude}[2]{\only<#1>{\includegraphics[width=\tmpwdth,clip,trim={0 0 0 2cm}]{#2}}}




% ------------------------------------------------------------------------------
\begin{frame}[label=aphylo-current]
	\frametitle{Phylogenetics Modeling Strategies}
	
	\begin{minipage}[m]{.3\linewidth}
		
		\begin{figure}
			\includegraphics[width=.9\linewidth]{phylo-model-overview-legend.pdf}
		\end{figure}
		
		% 		\uncover<5->{\large SNA could help us with\\\alert{\textbf{Exponential\\Random~Graph Models}}}
	\end{minipage}\hfill
	\begin{minipage}[m]{.69\linewidth}
		\mode<beamer>{
			\begin{figure}
				\phantom{\includegraphics<1>[width=.9\linewidth]{phylo-model-overview-1.pdf}}%
				\includegraphics<2>[width=.9\linewidth]{phylo-model-overview-1.pdf}%
				\includegraphics<3>[width=.9\linewidth]{phylo-model-overview-2.pdf}%
				\includegraphics<4>[width=.9\linewidth]{phylo-model-overview.pdf}
			\end{figure}
		}
		
		\mode<handout>{
			\begin{figure}
				\includegraphics[width=.9\linewidth]{phylo-model-overview.pdf}
			\end{figure}
		}
	\end{minipage}
	
\end{frame}

\section{Avoiding the Curse of dimensionality}

\begin{frame}[t]
	\usebeamertemplate{section intro}{}{}
	\textcolor{uscgold}{
		\Large {\bf Avoiding the Curse of dimensionality}
	}
\end{frame}

% ------------------------------------------------------------------------------
\begin{frame}[c]
	
	\Large If we wanted to build a model with 3 functions, we would need to estimate...\large
	\\\bigskip
	
	\begin{minipage}[t]{.40\linewidth}
		\centering
		\shadowbox{Full Markov Transition Matrix}\\\bigskip
		\uncover<1->{\includegraphics[width=.8\linewidth]{aphylo-ergm-eq1.png} \\
			\vfill 
		}
	\end{minipage}\hfill
	\uncover<2->{\begin{minipage}[t]{.50\linewidth}
			\vfill
			\begin{itemize}
				\item<2-> 512 parameters
				\item<3-> Finding this many parameters not easy.
				\item<4-> Even if you can, interpretation is awkward.
			\end{itemize}
	\end{minipage}}
	\vfill
	% 	\uncover<5->{Social Network Analysis may help us...}
\end{frame}



% -------------------------------------------------------------------------------
\begin{frame}[t]
	\frametitle{Exponential Random Graph Models (ERGMs)}
	
	\byside{.45}{Social Network}{%
		\includegraphics[height=3cm]{fig/adjmat-network.png}
	} \hfill %
	\begin{minipage}[t]{.45\linewidth}
		\bigskip
		\begin{itemize}
			\item<2-> Not about individual ties.
			\item<3-> Statistical inference on \textit{motifs} (triangles, dyads, homophily, etc.)
		\end{itemize}
		\bigskip
		\uncover<4->{Ultimately... \\
			\Large
			\textbf{ERGM} $\equiv$ \textbf{Modeling binary arrays}
			\normalsize}
		
	\end{minipage}
	
\end{frame}

% -------------------------------------------------------------------------------
\begin{frame}[t]
	\frametitle{Exponential Random Graph Models (ERGMs)}
	
	\byside{.45}{Social Network}{%
		\includegraphics[height=3cm]{fig/adjmat-network.png}
	} \hfill %
	\byside{.45}{Evolutionary Event}{%
		\includegraphics[height=3cm]{fig/adjmat-aphylo.png}
	}
	\vfill
	Social Networks are usually represented as \textbf{adjacency matrices}, and so can evolutionary events!
	
\end{frame}

% ------------------------------------------------------------------------------
\begin{frame}[c,label=aphylo-ergm-example]
	
	\Large If we wanted to build a model with 3 functions, we would need to estimate...\large
	\\\bigskip
	
	\begin{minipage}[t]{.40\linewidth}
		\centering
		\shadowbox{Full Markov Transition Matrix}\\\bigskip
		\includegraphics[width=.8\linewidth]{aphylo-ergm-eq1.png} \\
		\vfill 512 parameters
	\end{minipage}\hfill
	\begin{minipage}[t]{.19\linewidth}
		\centering 
		\uncover<3->{
			\includegraphics[width=.8\linewidth]{aphylo-ergm-eq2.png}\\
			Easier to fit \\
			Easier to interpret}	
	\end{minipage}\hfill
	\uncover<2->{\begin{minipage}[t]{.40\linewidth}		
			\centering
			\shadowbox{Sufficient statistics}\\\bigskip
			\includegraphics[width=.8\linewidth]{aphylo-ergm-eq2.png} \\
			\vfill 11 parameters (for example)
	\end{minipage}}
	\vfill
	\uncover<3->{\hfill\hfill\hyperlink{examples-geese-terms}{\beamerreturnbutton{term examples}}}
\end{frame}



%-------------------------------------------------------------------------------
\begin{frame}[c]
	\frametitle{Tree likelihoods: Felsenstein's Pruning algorithm}
	Also known as \textit{dynamic programming} or \textit{postorder tree traversal} 
	\begin{center}
	\mode<beamer>{%
		\includegraphics<1>[width=.6\linewidth]{geese-equation-likelihood-1.pdf}%
		\includegraphics<2->[width=.6\linewidth]{geese-equation-likelihood.pdf}
	}%
	\mode<handout>{%
		\includegraphics[width=.6\linewidth]{geese-equation-likelihood.pdf}
	}			
	\end{center}
\end{frame}

%-------------------------------------------------------------------------------
\begin{frame}[c]
	\frametitle{Predicting}
	\begin{center}
	\mode<beamer>{%
		\includegraphics<1->[width=.7\linewidth]{geese-equation-prediction.pdf}
	}%
	\mode<handout>{%
		\includegraphics[width=.7\linewidth]{geese-equation-prediction.pdf}
	}	
	\end{center}
	\vfill\hfill\uncover<2->{... I implemented this (and more) on \textbf{geese}}
\end{frame}

%-------------------------------------------------------------------------------
\begin{frame}[c]
	\centering
	\includegraphics[width=.25\linewidth]{fig/geese-logo.png}
	
	\scalebox{1.2}{{\color{usccardinal}\bfseries GE}ne functional {\color{usccardinal}\bfseries E}volution using {\color{usccardinal}\bfseries S}uffici{\color{usccardinal}\bfseries E}ncy}
	
	\vfill\hfill{... as part of \textbf{barry}, your to-go motif accountant \includegraphics[width=.075\linewidth]{fig/barry-logo.png}}
\end{frame}

\begin{frame}
	\frametitle{Computational Features of \textbf{geese}}
	\begin{minipage}[t]{.3\linewidth}
		\centering
		\uncover<2->{\shadowbox{Constrain support set}\\
			\includegraphics[width=1\linewidth]{aphylo-ergm-eq1.png}}
	\end{minipage}\hfill
	\begin{minipage}[t]{.3\linewidth}
		\centering
		\uncover<3->{\shadowbox{Support accounting}\\
			\includegraphics[width=1\linewidth]{ergm-computing.pdf}}
	\end{minipage}\hfill
	\begin{minipage}[t]{.3\linewidth}
		\centering
		\uncover<4->{\shadowbox{Reduced peeling sequence}\\
			\includegraphics[width=1\linewidth]{reduced-sequence.pdf}}
	\end{minipage}
	\vfill\hfill\uncover<5->{... how big can we go?}
	
\end{frame}

\begin{frame}
	\frametitle{Support size: Computational feasibility}
	Whether the likelihood is tracktable or not depends on how large is the \textbf{``largest event''}
	\begin{figure}
		\scalebox{.8}{
			\footnotesize
			%\includegraphics[width=.7\linewidth]{polytomies.pdf}
			\def\svgwidth{1\linewidth}
			\mode<handout>{\input{fig/polytomies.pdf_tex}}%
			\mode<beamer>{%
				\only<1>{\input{fig/polytomies-1.pdf_tex}}%
				\only<2>{\input{fig/polytomies-2.pdf_tex}}%
				\only<3->{\input{fig/polytomies.pdf_tex}}%
			}
		}
		%		\caption{\textit{Possible reconciliation states}. (a) Shows a fully reconciled tree in which each interior node has two decedents. (b) Illustrates a potentially reconciled tree, yet with a \textit{hard polytomy}. In that case, these five genes could be the result of a multifurcating speciation event. (c) Shows a \textit{soft polytomy}, which illustrates uncertainty about the best hypothesis connecting these genes.}
	\end{figure}
	\uncover<4->{\small(in practice, arrays up to 32 cells, i.e., 4.3 billion comb., are feasible.)}
\end{frame}


%\begin{frame}
%	\frametitle{What questions?}
%	
%	With this modeling framework, we could tackle, e.g.,
%	
%	\begin{itemize}
%		\item<2-> Potentially improve prediction accuracy.
%		\item<3-> Make inferences of the sort of:\\
%		\uncover<4->{"Function A or function B, which came first?"} \\
%		\uncover<5->{"When was subfunctionalization more likely to happen?"}\\
%		\uncover<6->{"Where functions A and B gained at the same time?"}
%		\item<7-> and much more...
%	\end{itemize}
%	
%\end{frame}

\section{Analyzing 77 experimentally annotated trees}

\begin{frame}[c]
	\usebeamertemplate{section intro}{}{}
	\textcolor{uscgold}{
		\Large Analyzing 77 experimentally annotated trees\\
		({\small Vega Yon, \textit{et al.}, WIP})
	}
\end{frame}

\begin{frame}[label=geese-vs-aphylo]
	\frametitle{Analyzing 77 experimentally annotated trees}
	Using data from Vega Yon et al. (2021), we re-estimated the models using \textbf{GEESE} and compared the results to those in the \textbf{aphylo} paper:
	
	\begin{itemize}
		\item<2-> 77 experimentally annotated trees
		\item<3-> We only used trees in which 
		$$2^{(\max{\mbox{Polytomy}} + 1)\times \mbox{nfuns}}< 0.5\times 10^9$$
		\noindent\uncover<4->{i.e., half billion}
		\item<5-> Both sets of models used "informative" priors.
		\item<6-> Both sets of were fitted using Robust Adaptive Metropolis (\textbf{fmcmc} R package).
	\end{itemize}
	\vfil\hfill\uncover<6->{\hyperlink{example-geese1}{\beamerbutton{more}}}
\end{frame}

\begin{frame}[fragile]
	\frametitle{Example of code (R)}
	
	After initializing a \texttt{geese} object named \texttt{model2fit}:\pause
	
	\begin{lstlisting}[language=R] 
		# For later use (see last two lines)
		term_overall_changes(model2fit, duplication = TRUE)
		term_overall_changes(model2fit, duplication = FALSE)
		
		# Couting how many genes change
		term_genes_changing(model2fit, duplication = TRUE)
		
		# Gain at duplication
		term_gains(model2fit, funs = 0:nfuns, duplication = TRUE)
		
		# Gain and loss at speciation
		term_gains(model2fit, funs = 0:nfuns, duplication = FALSE)
		term_loss(model2fit, funs = 0:nfuns, duplication = FALSE)
		
		# Constraining the support set
		rule_limit_changes(model2fit, id = 0, lb = 0, ub = 4, duplication = TRUE)
		rule_limit_changes(model2fit, id = 1, lb = 0, ub = 4, duplication = FALSE)\end{lstlisting}
	
\end{frame}

\begin{frame}
	\begin{figure}
%		\includegraphics[width=.45\linewidth]{geese-vs-aphylo-one-at-a-time.pdf}
		\caption{\textit{Performance as Mean Absolute Error (MAE)}. Each set of coordinates shows the value of 1 minus the MAE of \textbf{GEESE}, x-axis, and \textbf{aphylo} (y-axis). The statistics were computed using leave-one-out based on individual parameter estimates. Overall, GEESE performs better than aphylo in most cases.}
	\end{figure}
\end{frame}

%\begin{frame}[label=aphylo-pooled]
%	
%	\frametitle{The benefits of pooling data}
%	
%	Results from the \textbf{aphylo} model:
%	
%	% latex table generated in R 3.6.3 by xtable 1.8-4 package
%	% Tue Apr 28 11:51:17 2020
%	\begin{table}[tb]
%		\centering
%		\begin{tabular}{m{.14\linewidth}*{3}{m{.14\linewidth}<\centering}}
%			\toprule & \multicolumn{1}{c}{\textit{Pooled-data}} & \multicolumn{2}{c}{One-at-a-time} \\ \cmidrule(r){2-2}\cmidrule(r){3-4}
%			& Beta prior & Unif. prior & Beta Prior \\ 
%			\midrule
%			\textit{Pooled-data} \\
%			\hspace{2mm}Unif. prior & \cellcolor{blue!25}[-0.02,-0.01] & \cellcolor{blue!25}[-0.14,-0.10] & \cellcolor{blue!25}[-0.06,-0.03] \\ 
%			\hspace{2mm}Beta prior &  - & \cellcolor{blue!25}[-0.12,-0.09] & \cellcolor{blue!25}[-0.04,-0.01] \\ 
%			\textit{One-at-a-time} \\
%			\hspace{2mm}Unif. prior &  - & - & \cellcolor{red!25}[\hphantom{-}0.06,\hphantom{-}0.09] \\ 
%			\bottomrule
%		\end{tabular}
%		\caption[Differences in Mean Absolute Error]{Differences in Mean Absolute Error [MAE]. Each cell shows the 95\% confidence interval for the difference in MAE resulting from two methods (row method minus column method). Cells are color coded blue when the method on that row has a significantly smaller MAE than the method on that column; Conversely, cells are colored red when the method in that column outperforms the method in that row.  Overall, predictions calculated using the parameter estimates from \textit{pooled-data} predictions outperform \textit{one-at-a-time}.}
%		\label{tab:vs-accuracy}
%	\end{table}
%	
%\end{frame}

\begin{frame}
	\frametitle{Tapping into computational scalability}
	\pause
	\begin{figure}
		\includegraphics[width=.8\linewidth]{flock-plot.pdf}
		\caption{A dramatization of how a group of GEESE, i.e., a flock, looks like.}
	\end{figure}
	
\end{frame}


% ------------------------------------------------------------------------------
\begin{frame}[c]
	\centering
	\usebeamertemplate{section intro}{}{}
	\vspace{-1cm}
	\textbf{%
		\color{uscgold}
		\Large Applications of Statistical Computing in\\Complex Social and Biological Systems Modeling\large %
	}

	\textbf{
		\color{uscgold}
		George G Vega Yon \\
		\url{https://ggvy.cl} \\
		vegayon@usc.edu
	}
	
	
\includemedia[%
	width=.3\linewidth,%
	height=.3\linewidth,%
	addresource=fig/walking-dead.mp4,%
	transparent,
	%transparent player background
	activate=pageopen,
	passcontext,
	%show VPlayer's right-click menu
	flashvars={
		source=fig/walking-dead.mp4
		&loop=true
		% loop video
	}
]{}{VPlayer.swf}
	
	
	\begin{center}
	\scalebox{2}{\textcolor{uscgold}{Thank you!}} 
	\end{center}
\end{frame}


\renewcommand{\section}[2]{}%
\appendix

% ------------------------------------------------------------------------------
% ------------------------------------------------------------------------------
% ------------------------------------------------------------------------------
% ------------------------------------------------------------------------------
% ------------------------------------------------------------------------------

% ------------------------------------------------------------------------------
\changemode{handout}

 ------------------------------------------------------------------------------
\begin{frame}[label=go-functions-types]
	\frametitle{Genes and their Functions}
	
	Gene functions can be classified in three types:
	
	\def\tmpwidth{.9\linewidth}
	
	\begin{table}
		\begin{tabular}{*{3}{m{.31\linewidth}<{\centering}}}
			\onslide<2->\bf Molecular function & %
			\onslide<3->\bf Cellular component & %
			\onslide<4->\bf Biological process \\
			\onslide<2->\href{http://amigo.geneontology.org/amigo/term/GO:0005215}{Active transport GO:0005215}& %
			\onslide<3->\href{http://amigo.geneontology.org/amigo/term/GO:0004016}{Mitochondria GO:0004016} & %
			\onslide<4->\href{http://amigo.geneontology.org/amigo/term/GO:0060047}{Heart contraction GO:0060047} \\
			\onslide<2->\includegraphics[width=\tmpwidth]{Sodium-potassium_pump_and_diffusion.png} & %
			\onslide<3->\includegraphics[width=\tmpwidth]{640px-Animal_Cell-svg.png} & % 
			\onslide<4->\includegraphics[width=\tmpwidth]{Systolevs_Diastole.png}
		\end{tabular}
	\end{table}
	
	\vfill \hfill \hyperlink{geneontology}{\beamerreturnbutton{go back}}
	
\end{frame}


\begin{frame}[label=aphylo-goexample]
\frametitle{The Gene Ontology Project}

Example of GO term

\begin{table}
\footnotesize
\begin{tabular}{lm{.6\linewidth}}
\toprule
\textbf{Accession} & GO:0060047 \\
\textbf{Name} & heart contraction \\
\textbf{Ontology} & biological\_process \\
\textbf{Synonyms} & heart beating, cardiac contraction, hemolymph circulation \\
\textbf{Alternate} & IDs None \\
\textbf{Definition} & The multicellular organismal process in which the heart decreases in volume in a 
characteristic way to propel blood through the body. Source: GOC:dph \\
\bottomrule
\end{tabular}
\caption{Heart Contraction Function. source: \href{http://amigo.geneontology.org/amigo/term/GO:0060047}{amigo.geneontology.org}}
\end{table}%\pause

\vfill \hfill \hyperlink{geneontology}{\beamerreturnbutton{go back}}

\end{frame}

\begin{frame}[t, label=aphylo-good]
	\frametitle{Example: Molecular function in family PTHR1128}
	
	\begin{minipage}[m]{.39\linewidth}
		\small
		%	The AUC for this analysis is 0.91 and the Mean Absolute Error is 0.34
		%\uncover<1->{\textbf{Family: PTHR11258}}\\
		\uncover<1->{%
			%	\textbf{Type:} Molecular Function\\
			\textbf{Name:} 2'-5'-oligoadenylate synthetase activity\\
			\textbf{Desc:}  \href{http://amigo.geneontology.org/amigo/term/GO:0001730}{\alert{GO:0001730}} involved in the process of cellular antiviral activity (wiki on \href{https://en.wikipedia.org/wiki/Interferon}{\alert{interferon}}).
		}\\
		\uncover<2->{%
			\textbf{MAE:} 0.34 \\
			\textbf{AUC:} 0.91%
		}		
	\end{minipage}
	\begin{minipage}[m]{.6\linewidth}
		\centering
		
		\mode<beamer>{
			\only<1>{\includegraphics[width=.9\linewidth, clip, trim={0 0 0 2cm}]{example-trees-good1-parts-1.pdf}}%
			\only<2->{\includegraphics[width=.9\linewidth, clip, trim={0 0 0 2cm}]{example-trees-good1-parts-1b.pdf}}
		}
		\mode<handout>{
			\includegraphics[width=.9\linewidth, clip, trim={0 0 0 2cm}]{example-trees-good1-parts-1b.pdf}
		}
	\end{minipage}
	\vfill	
	\uncover<3->{
		\footnotesize \textbf{Note}: Prediction made using \textbf{aphylo} (Vega Yon and \textit{et al}, PLOS Comp. Bio 2021)
	} \hfill\hyperlink{phylo-legend}{\beamerreturnbutton{go back}}
	
\end{frame}

\begin{frame}[label=aphylo-data]
	\frametitle{Data: Phylogenetic trees}
	
	Sample of annotations (first 10 in a single tree, Phosphoserine Phosphatase [PTHR10000])
	
	\small
	
	\begin{table}[ht]
		\centering
		\begin{tabular}{rrll}
			\toprule
			Internal id & Branch Length & type & ancestor \\ 
			\midrule
			AN0 &  & S & LUCA \\ 
			AN1 & 0.06 & S & Archaea-Eukaryota \\ 
			AN2 & 0.24 & S & Eukaryota \\ 
			AN3 & 0.44 & S & Unikonts \\ 
			AN4 & 0.42 & S & Opisthokonts \\ 
			AN6 & 0.68 & D &  \\ 
			AN9 & 0.79 & S & Amoebozoa \\ 
			AN10 & 0.18 & D &  \\ 
			AN15 & 0.57 & S & Dictyostelium \\ 
			AN18 & 0.52 & S & Alveolata-Stramenopiles \\ 
			\bottomrule
		\end{tabular}
	\end{table}
	\vfill\hfill\hyperlink{phylo-table}{\beamerreturnbutton{go back}}
\end{frame}

\begin{frame}
	\frametitle{Data: Node type (events)}
	\begin{figure}
		\centering
		\includegraphics[width=.7\linewidth]{distribution-event-type.pdf}
	\end{figure}
	\vfill\hfill\hyperlink{phylo-table}{\beamerreturnbutton{go back}}
\end{frame}

\begin{frame}
	\frametitle{Data: Annotations (example)}
	
	This is the first 10 of $\sim$ 400,000 experimental annotations used:
	
	\footnotesize
	\begin{table}[ht]
		\centering
		\begin{tabular}{rllll}
			\toprule
			& Family & Id & GO term & Qualifier \\ 
			\midrule
			1 & PTHR12345 & HUMAN$|$HGNC=15756$|$UniProtKB=Q9H190 & GO:0005546 &  \\ 
			2 & PTHR11361 & HUMAN$|$HGNC=7325$|$UniProtKB=P43246 & GO:0016887 & CONTRIBUTES\_TO \\ 
			3 & PTHR10782 & MOUSE$|$MGI=MGI=3040693$|$UniProtKB=Q6P1E1 & GO:0045582 &  \\ 
			4 & PTHR23086 & ARATH$|$TAIR=AT3G09920$|$UniProtKB=Q8L850 & GO:0006520 &  \\ 
			5 & PTHR32061 & RAT$|$RGD=619819$|$UniProtKB=Q9EPI6 & GO:0043197 &  \\ 
			6 & PTHR46870 & ARATH$|$TAIR=AT3G46870$|$UniProtKB=Q9STF9 & GO:1990825 &  \\ 
			7 & PTHR15204 & MOUSE$|$MGI=MGI=1919439$|$UniProtKB=Q9Z1R2 & GO:0045861 &  \\ 
			8 & PTHR22928 & DROME$|$FlyBase=FBgn0050085$|$UniProtKB=Q9XZ34 & GO:0030174 &  \\ 
			9 & PTHR35972 & HUMAN$|$HGNC=34401$|$UniProtKB=A2RU48 & GO:0005515 &  \\ 
			10 & PTHR10133 & DROME$|$FlyBase=FBgn0002905$|$UniProtKB=O18475 & GO:0097681 &  \\ 
			\bottomrule
		\end{tabular}
	\end{table}
	\vfill\hfill\hyperlink{phylo-table}{\beamerreturnbutton{go back}}
\end{frame}

\begin{frame}
	\frametitle{Data: Experimental Annotations}
	\begin{figure}
		\centering
		\includegraphics[width=.7\linewidth]{distribution-annotation-type.pdf}
	\end{figure}
	\vfill\hfill\hyperlink{phylo-table}{\beamerreturnbutton{go back}}
\end{frame}


% ------------------------------------------------------------------------------
% ------------------------------------------------------------------------------
% ------------------------------------------------------------------------------
% ------------------------------------------------------------------------------


\begin{frame}[c,label=aphylo-table]
	\frametitle{Overview of Prediction Results: aphylo}
	
	\begin{minipage}[m]{.55\linewidth}
		\begin{table}
			\centering
			\scalebox{.64}{
				\begin{tabular}{%
						m{.24\linewidth}<\raggedright %
						>{\color<1>{usccardinal}}m{.24\linewidth}<\centering%
						>{\color<2>{usccardinal}}m{.24\linewidth}<\centering%
						>{\color<3>{usccardinal}}m{.24\linewidth}<\centering%
						>{\color<4>{usccardinal}}m{.24\linewidth}<\centering}
					\toprule
					& & \multicolumn{3}{c}{Type of Annotation} \\
					\cmidrule(r){3-5} %
					\phantom{\LARGE Cellular Component}& %
					\alt<1>{\LARGE Pooled}{Pooled} & %
					\alt<2>{\LARGE Molecular Function}{Molecular Function} & %
					\alt<3>{\LARGE Biological Process}{Biological Process} & %
					\alt<4>{\LARGE Cellular Comp.}{Cellular Component} \\ 
					\midrule
					\multicolumn{3}{l}{\hspace{-10pt}Mislabeling} \\
					$\psi_{01}$ & 0.23 & 0.18 & 0.09 & \multirow{11}{*}{\uncover<4->{\Huge ?}}\\ %0.66 \\ 
					$\psi_{10}$ & 0.01 & 0.01 & 0.01 & \\ %0.33 \\ 
					\multicolumn{3}{l}{\hspace{-10pt}Duplication Events} \\
					$\mu_{d01}$ & 0.97 & 0.97 & \hlcAlt{3}{0.10} & \\ %0.55 \\ 
					$\mu_{d10}$ & 0.52 & 0.51 & \hlcAlt{3}{0.03} & \\ %0.56 \\ 
					\multicolumn{3}{l}{\hspace{-10pt}Speciation Events} \\
					$\mu_{s01}$ & 0.05 & 0.05 & 0.05 & \\ % 0.37 \\ 
					$\mu_{s10}$ & 0.01 & 0.01 & 0.02 & \\ % 0.37 \\ 
					\multicolumn{3}{l}{\hspace{-10pt}Root node} \\
					$\pi$ & 0.79 & 0.71 & 0.88 & \\ %0.52 \\ \midrule
					Trees & 141 & 74 & 45 & 22 \\ 
					\multicolumn{3}{l}{\hspace{-10pt}Accuracy under the by-aspect model} \\
					AUC & - & \hlcAlt{2}{0.77} & \hlcAlt{3}{0.83} & \\ % 0.53 \\ 
					MAE & - & \hlcAlt{2}{0.34} & \hlcAlt{3}{0.26} & \\ % 0.50 \\ 
					\multicolumn{3}{l}{\hspace{-10pt}Accuracy under the pooled-data model} \\
					AUC & - & \hlcAlt{2}{0.77} & 0.75 & \\ % 0.75 \\ 
					MAE & - & \hlcAlt{2}{0.35} & 0.34 & \\ %0.37 \\ 
					\bottomrule
			\end{tabular}}
			%\caption[Parameter estimates comparing pooled-data vs by-type]{MCMC estimates for experimentally annotated trees. The first column shows the estimates under the pooled-data model in \ref{tab:pooled-experimentally-annotated}, while the following three columns report the estimates obtained when fitting the model using a pooled-data approach, but doing so by type of annotation. Readers should be aware that the estimation process of the fourth column, \textit{cellular component}, did not fully converge, likely due to sparsity of annotations within that category.}
			\label{tab:by-aspect-estimates}
		\end{table}
	\end{minipage}
	\hfill
	\begin{minipage}[m]{.44\linewidth}
		Previously, joint estimates out-performed one-at-a-time\pause
		\begin{itemize}
			\item \textbf{Molecular Function} No change.\pause
			\item \textbf{Biological Process} Significantly better.\pause
			\item \textbf{Cellular Component} Does not converge.\pause
		\end{itemize}
		
		
		
		\small
		\begin{table}
			\begin{tabular}{%
					m{.22\linewidth}<\centering m{.03\linewidth}<\centering%
					m{.22\linewidth}<\centering m{.03\linewidth}<\centering%
					m{.25\linewidth}<\centering%
				}
				Molecular Function & $\neq$ & Biological Process &  ? & Cellular Component
			\end{tabular}
		\end{table}
		\normalsize
		
		\only<1->{\vfill\hfill\hyperlink{phylo-data}{\beamergotobutton{data}}}
		\only<1->{\vfill\hfill\hyperlink{aphylo-results-overview}{\beamergotobutton{go back}}}
		
	\end{minipage}
	
	
	
\end{frame}


%\begin{frame}[t]
%	\frametitle{Simulation Study}
%	
%	\begin{minipage}[t]{.3\linewidth}
%		\begin{enumerate}
%			\item Higher convergence rate
%			\item Smaller bias
%			\item Higher power
%			\item Smaller type I error
%			\item \alert{\bf Elapsed time}
%		\end{enumerate}
%		\vfill\hyperlink{ergmito-results-brief}{\beamerreturnbutton{return}}
%	\end{minipage}
%	\hfill\begin{minipage}[t]{.65\linewidth}%\vspace{-.75cm}
%		\begin{figure}
%			\centering
%			%	\caption[Distribution of Elapsed Time]{\label{fig:elapsedtime}Distribution of elapsed time (in seconds) for the estimation process for MC-MLE (ERGM) versus MLE (using \ergmito{}). Overall, the MLE implementation is orders of magnitude faster compared to the time required by the MC-MLE implementation to do the parameter estimation.}
%			\small
%			\def\svgwidth{.9\linewidth}
%			\input{fig/time.pdf_tex}
%		\end{figure}
%	\end{minipage}
%\end{frame}

% ------------------------------------------------------------------------------
%\begin{frame}[c,label=ergmito-teams-data]
%	\frametitle{Featured : Small Teams}
%	
%	\begin{minipage}[t]{.38\linewidth}
%		\small
%		\noindent \textbf{Sample}
%		\pause
%		\begin{itemize}
%			\item 31 mixed-gender teams (4-5 members).\pause
%			\item University students.\pause
%			\item Don't know each other.\pause
%			\item At least one female/male per team.\pause
%		\end{itemize}
%		\noindent \textbf{Experiment}\pause
%		\begin{itemize}
%			\item Complete 1 hour of group tasks.\pause
%			\item Captured network data using name generator survey: \textit{Who did you go to for advice, information or help to complete the group task?}
%		\end{itemize}
%	\end{minipage}
%	\hfill
%	\begin{minipage}[t]{.6\linewidth}
%		\centering
%		\uncover<9->{
%			\begin{figure}
%				\includegraphics[width=.95\linewidth]{ci-ergm-all-nets.pdf}
%			\end{figure}
%			\uncover<10->{\Large Is Gender Homophily a feature of these graphs?}
%		}
%	\end{minipage}
%	\vfill\hfill\hyperlink{ergmito-applied}{\beamerreturnbutton{go back}}
%\end{frame}

%% ------------------------------------------------------------------------------
%\begin{frame}[label=aphylo-current]
%	\frametitle{Phylogenetics Modeling Strategies}
%	
%	\begin{minipage}[m]{.3\linewidth}
%		
%		\begin{figure}
%			\includegraphics[width=.9\linewidth]{phylo-model-overview-legend.pdf}
%		\end{figure}
%	\end{minipage}\hfill
%	\begin{minipage}[m]{.69\linewidth}
%		\mode<beamer>{
%			\begin{figure}
%				\phantom{\includegraphics<1>[width=.9\linewidth]{phylo-model-overview-1.pdf}}%
%				\includegraphics<2>[width=.9\linewidth]{phylo-model-overview-1.pdf}%
%				\includegraphics<3>[width=.9\linewidth]{phylo-model-overview-2.pdf}%
%				\includegraphics<4>[width=.9\linewidth]{phylo-model-overview.pdf}
%			\end{figure}
%		}
%		
%		\mode<handout>{
%			\begin{figure}
%				\includegraphics[width=.9\linewidth]{phylo-model-overview.pdf}
%			\end{figure}
%		}
%	\end{minipage}
%	
%	\vfill\hfill\hyperlink{current-research}{\beamerreturnbutton{go back}}
%	
%\end{frame}

%% ------------------------------------------------------------------------------
%\begin{frame}[c,label=aphylo-ergm-example]
%	
%	\Large If we wanted to build a model with 3 functions, we would need to estimate...\large
%	\\\bigskip
%	
%	\begin{minipage}[t]{.40\linewidth}
%		\centering
%		\shadowbox{Full Markov Transition Matrix}\\\bigskip
%		\uncover<2->{\includegraphics[width=.8\linewidth]{aphylo-ergm-eq1.png} \\
%			\vfill 512 parameters
%		}
%	\end{minipage}\hfill
%	\begin{minipage}[t]{.19\linewidth}
%		\centering 
%		\uncover<4->{
%			\includegraphics[width=.8\linewidth]{aphylo-ergm-eq2.png}\\
%			Easier to fit \\
%			Easier to interpret}	
%	\end{minipage}\hfill
%	\begin{minipage}[t]{.40\linewidth}		
%		\centering
%		\shadowbox{Sufficient statistics}\\\bigskip
%		\uncover<3->{\includegraphics[width=.8\linewidth]{aphylo-ergm-eq2.png} \\
%			\vfill 5 parameters}
%	\end{minipage}
%	\uncover<4->{\vfill\hfill\hyperlink{current-research}{\beamerreturnbutton{numeric example}}}
%	
%\end{frame}


%-------------------------------------------------------------------------------
\begin{frame}[c, label=barry-pkg]
	\frametitle{Barry: your go-to \textit{motif} accountant}
	\centering
	\begin{minipage}[m]{.33\linewidth}
		\small
		\begin{itemize}
			\item<1-> Sparse matrix represented using double hashmaps (fast row/column access).
			\item<2-> Template implementation for flexible weights and metadata.
			%	\item<3-> Full enumeration of array powersets.
			\item<3-> Fast counting using change statistics (\tiny Ch. 4\normalsize).
			\item<4-> Calculation of support for sufficient stats.
		\end{itemize}
		\only<1->{\url{https://USCbiostats.github.io/binaryarrays}}
	\end{minipage}\hfill
	\begin{minipage}[m]{.65\linewidth}
		\begin{figure}
			\begin{minipage}[m]{.49\linewidth}
				\uncover<5->{\includegraphics[width=.99\linewidth]{barray-phylo.png}}
			\end{minipage}
			\begin{minipage}[m]{.49\linewidth}
				\uncover<6->{\includegraphics[width=.99\linewidth]{barray-network.png}}
			\end{minipage}
			\uncover<5->{\caption{Screenshots from the project's website on GitHub.}}
		\end{figure}
	\end{minipage}
\end{frame}


% ------------------------------------------------------------------------------
\begin{frame}[t, label=aphylo-ergm-table]
	\frametitle{What Drives Evolution}
	
	Imagine that we have 3 functions (rows) and that each node has 2 siblings (columns)
	
	\mode<beamer>{
		\begin{table}
			\begin{tabular}{llcc}
				\toprule
				& & \multicolumn{2}{c}{\bf Transitions to} \\
				& & Case 1 & Case 2 \\ \cmidrule(r){3-4}
				\multicolumn{2}{r}{\textbf{Parent} $\begin{array}{c}\mbox{A} \\ \mbox{B} \\ \mbox{C}\end{array}\left[\begin{array}{c}0 \\ 1 \\ 1\end{array}\right]$} & 
				$\left[\begin{array}{cc} %
					0 & \nhlc{1-2}{1}\hlc{3-5}{1}\nhlc{6-}{1} \\ %
					1 & \nhlc{1-3}{0}\hlc{4-5}{0}\nhlc{6-}{0} \\ %
					1 & \nhlc{1-3}{0}\hlc{4}{1}\nhlc{5-}{1} %
				\end{array}\right]$ & 
				$\left[\begin{array}{cc} %
					0 & \nhlc{-2}{1}\nhlc{4}{1}\hlc{3}{1}\hlc{5}{1}\nhlc{6-}{1} \\ %
					\nhlc{-5}{1}\hlc{6}{1} & \nhlc{-4}{0}\hlc{5}{0}\nhlc{6-}{0}\\ %
					\nhlc{-4}{0}\hlc{5}{0}\nhlc{6-}{0} & \nhlc{1-5}{1}\hlc{6}{1}%
				\end{array}\right]$ \pause \\ \midrule 
				\multicolumn{3}{l}{\textbf{Sufficient statistics}} \pause \\ 
				& \# Gains & 1 & 1 \pause \\
				& Only one offspring changes (yes/no) & 1 & 0 \pause \\
				& \# Changes (gain+loss) & 2 & 3 \pause \\
				& Subfunctionalization (yes/no) & 0 & 1 \\ \bottomrule
			\end{tabular}
		\end{table}
	}
	\mode<handout>{
		\begin{table}
			\begin{tabular}{llcc}
				\toprule
				& & \multicolumn{2}{c}{\bf Transitions to} \\
				& & Case 1 & Case 2 \\ \cmidrule(r){3-4}
				\multicolumn{2}{r}{\textbf{Parent} $\begin{array}{c}\mbox{A} \\ \mbox{B} \\ \mbox{C}\end{array}\left[\begin{array}{c}0 \\ 1 \\ 1\end{array}\right]$} & 
				$\left[\begin{array}{cc} %
					0 & 1 \\ %
					1 & 0 \\ %
					1 & 1 %
				\end{array}\right]$ & 
				$\left[\begin{array}{cc} %
					0 & 1 \\ %
					1 & 0\\ %
					0 & 1%
				\end{array}\right]$ \\ \midrule 
				\multicolumn{3}{l}{\textbf{Sufficient statistics}} \\ 
				& \# Gains & 1 & 1  \\
				& Only one offspring changes (yes/no) & 1 & 0 \\
				& \# Changes (gain+loss) & 2 & 3 \\
				& Subfunctionalizations (yes/no) & 0 & 1 \\ \bottomrule
			\end{tabular}
		\end{table}
	}
	
	\vfill\hfill\hyperlink{aphylo-ergm-example}{\beamergotobutton{return}}
	%	\pause
	%	Modeling the full Markov transition matrix would take $2^3 \times 2^6 = 512$ parameters.
	
\end{frame}


%\begin{frame}[allowframebreaks]
%	\frametitle{References}
%	% \bibliographystyle{apacite}
%	% \bibliography{bibliography.bib}
%	\printbibliography
%\end{frame}

\begin{frame}[label=examples-geese-terms]
	\def\fwidth{.55\linewidth}
	\begin{table}
		\begin{tabular}{m{.2\linewidth}<\centering m{.2\linewidth}m{.4\linewidth}}
			\toprule
			Representation & Description & Definition  \\ \midrule
			\includegraphics[width=\fwidth]{fig/term-gain.png} & %
			Gain of function & $(1 - x_p)\sum_{n:n\in Off}x_n$  \\
			\includegraphics[width=\fwidth]{fig/term-loss.png} & %
			Loss of function & $x_p\sum_{n:n\in Off}(1 - x_n)$  \\
			\includegraphics[width=\fwidth]{fig/term-subfun.png} & %
			Subfunctionalization & $x_p^kx_p^j\sum_{n\neq m}x_n^k(1-x_n^j)(1-x_m^k)x_m^j$  \\
			\includegraphics[width=\fwidth]{fig/term-neofun.png} & %
			Neofunctionalization & $x_p^k(1 - x_p^j)\sum_{n\neq m}x_n^k(1-x_n^j)(1-x_m^k)x_m^j$ \\
			\includegraphics[width=\fwidth]{fig/term-longest.png} & %
			Longest branch gains & $(1-x_p^k)\isone{x_m^k : m=\mbox{argmax}_n\mbox{blength}_n}$ \\
			\bottomrule
		\end{tabular}
		\caption{Example of sufficient statistics for evolutionary transitions.}
	\end{table}
	\vfill\hfill\hyperlink{aphylo-ergm-example}{\beamerreturnbutton{go back}}
\end{frame}


\begin{frame}[c,label=example-geese1]
	\frametitle{Example: Simple model with two functions}
	To illustrate, we will \textbf{simulate} and then \textbf{estimate} the parameters for the following process:
	\begin{enumerate}
		\item<2-> 100 genes on a simulated phylogenetic tree.
		\item<2-> Two functions, \textbf{0} and \textbf{1},
		\item<3-> {\color{blue}\textbf{Function 0}} is likely to be gain \textbf{gained} \textbf{at a dupl. event},
		\item<3-> {\color{red}\textbf{Function 1}} is gained as \textbf{neofunctionalization} ({\color{blue}\textbf{from 0}}) \textbf{at a dupl. event},
		\item<4-> There is a higher chance of \textbf{changes at duplication} (explicit).
		\item<4-> Root node starts off without either function (i.e. prob $\to$ 0).
	\end{enumerate}
	\vfill
	\uncover<5->{We will fit the model using Robust Adaptive Metropolis with a logistic prior centered at 0 with scale 2.} %
	\hfill\uncover<5->{\hyperlink{geese-vs-aphylo}{\beamerreturnbutton{go back}}}
	
\end{frame}


\begin{frame}[c]
	\frametitle{Example: Simple model with two functions}
	\framesubtitle{posterior distributions}
	\begin{figure}
		\includegraphics[width=.8\linewidth]{fig/trace-gain-neo.png}
		\caption{MCMC Trace of the functional gain of 0, neofunctionalization (1 from 0), and change rate (by event type).}
	\end{figure}
\end{frame}


\begin{frame}[c]
	\frametitle{Example: Simple model with two functions}
	\framesubtitle{posterior distributions (contd')}
	\begin{figure}
		\includegraphics[width=.8\linewidth]{fig/trace-root.png}
		\caption{MCMC Trace of root parameters. The true population parameters are $(\theta_{root0}, \theta_{root1}) = (-10.0, -10.0)$. Root node probabilities are always hard to get.}
	\end{figure}
\end{frame}


%-------------------------------------------------------------------------------
\begin{frame}
	\frametitle{Simulation Study}
	
	\begin{minipage}[c]{.40\linewidth}
		Repeated this experiment 5,000 times:
		\begin{itemize}
			\item MCMC for fitting.
			\item RAM kernel.
			\item Logistic prior at zero with scale two.
			\item Each tree took $<$ 1min estimation.
		\end{itemize}
	\end{minipage}\hfill
	\begin{minipage}[c]{.59\linewidth}
		\uncover<2->{\begin{figure}    
				\centering
				\caption{Distribution of parameter estimates from 5,000 phylo trees w/ 100 leafs.}
				\includegraphics[width=.85\linewidth]{fig/aphylo-sim-study.pdf}
		\end{figure}}
	\end{minipage}
	
\end{frame}


\end{document}

